### ISSUE TITLE
`LostInstance` false-positive

### ISSUE THREAD

**Issue by boris-petrov** [checkstyle-addons/checkstyle-addons#4]
```java
public class Test {
        public String foo(Object x) {
                return switch (x) {
                        case String y -> {
                                yield new String("42");
                        }
                        default -> "42";
                };
        }
}
```

The `yield` statement causes `LostInstance` (which is a bug).

cc @tsjensen

**Comment by tsjensen** [comment]
Ah, that's a Java 13 construct. So far, I have done almost no testing with Java greater than 8. We should have such tests in the CI, really.  
Thanks for letting me know! I will think about it. Most of our checks are not affected much by the changes in newer Java, but still.  
It will be a while until the next Checkstyle Addons "sprint" though. Pull requests welcome!

**Comment by tsjensen** [comment]
I made the first few baby steps on branch [issue4](https://github.com/checkstyle-addons/checkstyle-addons/tree/issue4).

The idea is like this:

- Pimp the build system to know a new source set `test17` which will simply contain unit tests which are executed using Java 17. Java 17 is available because we need it for running Gradle anyway.
- Inside that source set, we've got a new test case with the code from above.
- We execute the test case against the newest variant of Checkstyle (which is the `default` depConfig). Older versions are not so interesting because they mostly wouldn't support the newer Java required.

The build system is very complex due to the chaotic nature of Checkstyle, so setting this up in the build is the true challenge, after which the fix itself will be a walk in the park. 😸
